Source/OptionalDataSource.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  DataSourceEditing.swift
+//  JSQDataSourcesKit
+//
+//  Created by Panagiotis Sartzetakis on 12/09/2016.
+//  Copyright © 2016 Hexed Bits. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol DataSourceTableEditingProtocol {
```

Comment:
Thinking about this... 🤔 

Since these editing methods are specific to `tableView` -- that is, there is no equivalent for `collectionView` ... maybe we don't need this protocol at all? I think we can just have the concrete controller below.